src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java

```diff
@@ -22,25 +22,57 @@
 import java.io.File;
 import java.io.IOException;
 import java.io.OutputStream;
+import java.lang.reflect.Field;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.ArrayDeque;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Deque;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+
+import com.google.common.collect.Lists;
 import com.puppycrawl.tools.checkstyle.AbstractAutomaticBean.OutputStreamOptions;
 import com.puppycrawl.tools.checkstyle.Checker;
 import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
 import com.puppycrawl.tools.checkstyle.MetadataGeneratorLogger;
 import com.puppycrawl.tools.checkstyle.TreeWalker;
 import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.checks.TodoCommentCheck;
+import com.puppycrawl.tools.checkstyle.site.SiteUtil;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+import org.apache.maven.doxia.macro.MacroExecutionException;
 
 /** Class which handles all the metadata generation and writing calls. */
 public final class MetadataGeneratorUtil {
 
+    /** Regular expression for detecting ANTLR tokens(for e.g. CLASS_DEF). */
+    private static final Pattern TOKEN_TEXT_PATTERN = Pattern.compile("([A-Z_]{2,})+");
+
+    /**
+     * Modules whose metadata is generated by functionality shared with SiteUtil, not
+     * by JavadocMetadataScraper.
+     * Until <a href="https://github.com/checkstyle/checkstyle/issues/13626">#13626</a>
+     */
+    private static final Set<String> MODULES_GENERATED_BY_NEW_FUNCTIONALITY = Set.of(
+        "TodoCommentCheck.java"
```

Comment:
to avoid dealing with such list, that cause conflicts .... annoying.
can we have some code that check if template has "<macros " keyword we activate few functionality ?

Reply:
I do not care about performance here, so simple load of while file as sting to memory  and doing "contains" it is good.

Reply:
Uhh, I got something working but metadata generation is 10 seconds slower. Is that alright or do we aim lower?

Reply:
@stoyanK7 , what about this ?

Reply:
Done as well.